src/cascadia/UnitTests_Control/ControlInteractivityTests.cpp

```diff
@@ -924,27 +919,48 @@ namespace ControlUnitTests
                                       0, // timestamp
                                       modifiers,
                                       cursorPosition1.to_core_point());
+        VERIFY_ARE_EQUAL(0, expectedOutput.size(), L"Validate we drained all the expected output");
+
+        Log::Comment(L" --- Scroll up, click the terminal. We shouldn't get any event. ---");
+        core->UserScrollViewport(10);
+        VERIFY_IS_GREATER_THAN(core->ScrollOffset(), 0);
+
+        // Viewport is now above the mutable viewport, so the mouse event
+        // straight up won't be sent to the terminal.
+
+        expectedOutput.push_back(L"sentinel"); // Clearly, it won't be this string
+        interactivity->PointerPressed(leftMouseDown,
+                                      WM_LBUTTONDOWN, //pointerUpdateKind
+                                      0, // timestamp
+                                      modifiers,
+                                      cursorPosition0.to_core_point());
+        // Flush it out.
+        conn->WriteInput(L"sentinel");
+        VERIFY_ARE_EQUAL(0, expectedOutput.size(), L"Validate we drained all the expected output");
 
-        Log::Comment(L"Switch to alt buffer");
+        Log::Comment(L" --- Switch to alt buffer ---");
         term.Write(L"\x1b[?1049h");
+        auto returnToMain = wil::scope_exit([&]() { term.Write(L"\x1b[?1049h"); });

```

Comment:
we could just give these all Method-level isolation..!

Reply:
after watching VS choke on going in and out of debugging on method-level isolation all day, _nah this is close enough_